client/signup/controller.js

```diff
@@ -233,6 +234,21 @@ export default {
 
 		store.set( 'signup-locale', localeFromParams );
 
+		/**
+		 * The experiment is only loaded on the onboarding flow
+		 * If user is logged out we load the experiment
+		 * If user is logged in we load the experiment only if the user has no sites
+		 * More info: pbxNRc-3xO-p2
+		 */
+		const isNewUser = ! getCurrentUserSiteCount( context.store.getState() );
+		initialContext.isSignupSurveyActive = false;
+		const isOnboardingFlow = flowName === 'onboarding';
+		if ( isOnboardingFlow && ( ! userLoggedIn || ( userLoggedIn && isNewUser ) ) ) {
+			const experiment = await loadExperimentAssignment(
+				'calypso_signup_onboarding_site_goals_survey'
+			);
+			initialContext.isSignupSurveyActive = experiment.variationName === 'variant';
+		}
 		next();
 	},
```

Comment:
@aaronyan Please note the experiment loading logic.
The experiment is loaded `on load` of the `/start`page given that the above conditions are met

Reply:
Looks good to me. The only thing is to change the `variant` to `treatment`!

```suggestion
		/**
		 * The experiment is only loaded on the onboarding flow
		 * If user is logged out we load the experiment
		 * If user is logged in we load the experiment only if the user has no sites
		 * More info: pbxNRc-3xO-p2
		 */
		const isNewUser = ! getCurrentUserSiteCount( context.store.getState() );
		initialContext.isSignupSurveyActive = false;
		const isOnboardingFlow = flowName === 'onboarding';
		if ( isOnboardingFlow && ( ! userLoggedIn || ( userLoggedIn && isNewUser ) ) ) {
			const experiment = await loadExperimentAssignment(
				'calypso_signup_onboarding_site_goals_survey'
			);
			initialContext.isSignupSurveyActive = experiment.variationName === 'treatment';
		}
		next();
	},
```